experiment/simulation/js/main.js

```diff
@@ -1 +1,288 @@
+		output_mot.innerHTML = this.value;
+		dataNum = Number(document.getElementById("motion").value);
+		await restart();
+	};
+
+	function curvedArea(ctx, e, gradX, gradY)
+	{
+		ctx.bezierCurveTo(e[0], e[1] += gradY, e[0] += gradX, e[1] += gradY, e[0] += gradX, e[1]);
+		ctx.bezierCurveTo(e[0] += gradX, e[1], e[0] += gradX, e[1] -= gradY, e[0], e[1] -= gradY);
+	}
+
+	function drawGround(ctx, ground)
+	{
+		ctx.save();
+		ctx.fillStyle = "pink";
+		ctx.beginPath();
+		ctx.moveTo(ground[0][0], ground[0][1]);
+
+		for(let i = 0; i < ground.length; ++i)
+		{
+			const next = (i + 1) % ground.length;
+			ctx.lineTo(ground[next][0], ground[next][1]);
+		}
+
+		ctx.closePath();
+		ctx.fill();
+		ctx.stroke();
+		ctx.restore();
+	}
+
+	let height = 400;
+	let dataNum = 1;
+
+	let data = [];
+	const file = `./data/groundData${dataNum}.json`;
+	await fetch(file)
+		.then(response => response.json())
+		.then(jsonResponse => {data = {...jsonResponse}});
+
+	const canvas = document.getElementById("main");
+	canvas.width = 1200;
+	canvas.height = 600;
+	canvas.style = "border:3px solid";
+	const ctx = canvas.getContext("2d");
+
+	const fill = "#A9A9A9";
+	const lineWidth = 1.5;
+
+	const fps = 10;
+	const scale = 25;
+
+	const defY = 100;
+	const startL = [120, 370, 770];
+	const startR = [270, 670, 1070];
+
+	let upL = {...startL};
+	let upR = {...startR};
+	upL[1] = startL[1] + (startR[1] - startL[1]) / 3;
+	upR[1] = startR[1] - (startR[1] - startL[1]) / 3;
+	upL[2] = startL[2] + (startR[2] - startL[2]) / 3;
+	upR[2] = startR[2] - (startR[2] - startL[2]) / 3;
+	const thickness = 10;
+
+	let bldg = [
+		[[upL[0], defY], [upR[0], defY], [startR[0], defY + height], [startL[0], defY + height]],
+		[[upL[1], defY], [upR[1], defY], [startR[1], defY + height], [startL[1], defY + height]],
+		[[upL[2], defY], [upR[2], defY], [startR[2], defY + height], [startL[2], defY + height]]
+	];
+
+	let ground = [
+		[startL[0] - 50, defY + height + 40],
+		[startL[0], defY + height - 40],
+		[startR[2] + 50, defY + height - 40],
+		[startR[2], defY + height + 40],
+	];
+	
+	let layer2 = [
+		{...ground[0]},
+		[ground[0][0], defY + height + 40 + thickness],
+		[startR[2] + thickness, defY + height + 40 + thickness],
+		[ground[2][0] + thickness, defY + height - 40 + thickness],
+		{...ground[2]},
+		{...ground[3]},
+	];
+
+	let keys = Object.keys(data);
+	let ind = 0;
+	let prev = keys.length - 1;
+
+	function draw()
+	{
+		ctx.clearRect(0, 0, canvas.width, canvas.height);
+		ctx.fillStyle = fill;
+		ctx.lineWidth = lineWidth;
+		ctx.lineCap = "round";
+		ctx.lineJoin = "round";
+
+		ground.forEach(g => {
+			g[0] += scale * data[keys[ind]];
+		});
+		layer2.forEach(l => {
+			l[0] += scale * data[keys[ind]];
+		});
+
+		drawGround(ctx, ground);
+		drawGround(ctx, layer2);
+
+		ground.forEach(g => {
+			g[0] -= scale * data[keys[ind]];
+		});
+		layer2.forEach(l => {
+			l[0] -= scale * data[keys[ind]];
+		});
+
+		for(let k = 0; k < 3; ++k)
```

Comment:
Please either put some comments on different parts of this function of consider breaking it into 2. This is a very arge function and the code isn't easy to understand.